### DISCUSSION TITLE
Expand auth record in auth-with-password

### DISCUSSION THREAD

**Discussion by bmingles** [pocketbase/pocketbase#3924] [Q&A] [answered]
I have an auth collection with a 1 to many relationship to a subscriptions collection via a `subs` field. My subscriptions collection then has a 1 to 1 relation with my products collection which has a sku field. I want to expand the `subs` field to include the sku when logging in.

I've attempted:

`/api/collections/myUsers/auth-with-password?expand=subs.product.sku`

also

`/api/collections/myUsers/auth-with-password?expand=subs,subs.product,subs.product.sku`

Neither of these worked. The result always just shows an array of subscription ids in the `subs` field. Not sure what I'm missing here.

**Comment by ganigeorgiev** [comment, marked as answer]
`expand` works only for relation fields, aka. if I understand your described structure something like `expand=subs.product` should work as long as the request satisfies the View API rules of the relations.

If you are not able to make it work, feel free to paste your collections configuration as json (_Admin UI > Settings > Export collections_) and I'll try to test it.

**Reply by bmingles** [reply]
Looks like api rules was my issue. Expansion works once I updated those.

Is it possible to trim down the fields in the expanded sections? Something like:

```
{
    "expand": {
      "subs": [
        {
          "expand": {
            "product": {
              "sku": "ProductA",
            }
          },
        },
        {
          "expand": {
            "product": {
              "sku": "ProductB",
            }
          },
        },
      ]
    },
  },

**Reply by ganigeorgiev** [reply]
If you need to limit the returned fields you can pass the `fields` query parameter, but note that for the auth-with-password endpoint it needs to correspond to the returned json. Here is an example with the JS SDK:

```js
await pb.collection("example").authWithPassword("test@example.com", "1234567890", {
    expand: "subs.product",
    // *                                     - return token, record and any other root level keys
    // record.*                              - return any root level record object keys (needed since otherwise the record object will get replaced by the next line)
    // record.expand.subs.expand.product.sku - return only the sku within the product object
    fields: "*,record.*,record.expand.subs.expand.product.sku"
})
```

**Reply by bmingles** [reply]
Lovely. I was missing the `record` piece. Thank you for your help.
